kernel/examples/read-table-multi-threaded/src/main.rs

```diff
@@ -277,23 +215,16 @@ fn do_work(
                 let root_url = Url::parse(&scan_state.table_root).unwrap();
 
                 // get the selection vector (i.e. deletion vector)
-                let mut selection_vector = scan_file.dv_info.map(|info| {
-                    let descriptor = DeletionVectorDescriptor::new(
-                        info.storage_type,
-                        info.path_or_inline_dv,
-                        info.offset,
-                        info.size_in_bytes,
-                        info.cardinality,
-                    );
-                    delta_kernel::scan::selection_vector(engine_interface, &descriptor, &root_url)
-                        .unwrap()
-                });
+                let mut selection_vector = scan_file
+                    .dv_info
+                    .get_selection_vector(engine_interface, &root_url)
+                    .unwrap();
 
                 // build the required metadata for our parquet handler to read this file
                 let location = root_url.join(&scan_file.path).unwrap();
                 let meta = FileMeta {
                     last_modified: 0,
-                    size: scan_file.size,
+                    size: scan_file.size as usize,
```

Comment:
it'll be _giant_. i think this works because they're the same # of bits but it just reinterprets them so the sign bit makes it huge.

really we should fix this by having size be a `usize` in the struct (or some other reasonable unsigned type). i made #174 to follow up on this